docs/DeveloperGuide.md

```diff
@@ -1,25 +1,91 @@
+
+Most deliverymen would want an overall outlook of completed deliveries, so that they would
+be able to see and judge exactly how much they have earned over the course of their delivery spree.
+
+The primary class involved in executing this function is `Deliveryman`.
+
+Whenever a `Delivery` is marked as completed, the `Ui` class will call
+the `completeDelivery()` method which accepts a delivery number (the index of the
+delivery), and a Deliveryman object.
+
+The `completeDelivery()` in Delivery then subsequently calls another `completeDelivery` method, a
+instance method of the `Deliveryman` class. This method will add the completed delivery to a
+list of completed deliveries (`records`) attributed to the deliveryman.
+
+When the `record` command is given, the `showRecords()` method is called which accepts the
+`Deliveryman` object. This method will simply print a formatted list of all the items in `records`
+
+This sequence of program flow was chosen because it would still allow for the proper abstraction of the classes.
+In this particular case, the `Ui` class is only handling the User Interface aspect of code execution. Similarly,
+the `Delivery` class is not especially involved in the `Deliveryman`'s `records`. The code directly changing the deliveryman's
+`records` is present only in the `Deliveryman` class.
+
+The general interaction between the abovementioned classes are briefly illustrated in the sequence diagram
+shown below. Note the two different `completeDelivery()` methods that are called here. Despite having
+similar names, their impacts are felt by different classes and maintains the abstraction between classes
+
+![Retrieve Records](diagrams/RetrieveRecords.png "Retrieving Records Sequence Diagram")
+
+### [Implementation] : Feature [2]
+
+### [Implementation] : List Assigned Deliveries
+
+All deliverymen would want have a list of deliveries that are assigned to them so that they would be able to get an 
+overview delivery jobs that have on that day.
+
+The primary class involved executing this function is `Delivery`.
+
+Whenever the user request for the delivery list, the `Ui` class will call the `showDeliveryList()` method. 
+
+The `showDeliveryList()` in Delivery then subsequently retrieve the list of deliveries from the static variable 
+`DeliveryList.deliveries`. This static variable contains the list of completed and uncompleted deliveries. 
+
+Having obtained the list of deliveries, the program will iterate through this list of `Delivery` objects and calls the 
+`toString()` method that has been overridden in the `Delivery` class. The `toString()` method has been overridden 
+specifically to return a formatted string that includes `deliveryID`,  `DeliveryStatusSymbol`, `Address` and `Recipient`
+information for each delivery job. 
+
+The sequence of program flow was chosen as it allows for proper abstraction of the classes. The `Ui` class will only 
+handle the User Interface aspect of the code execution. 
```

Comment:
I like how you guys clearly explain the sequence diagram before it is shown. I also like how you guys include just enough detail so that a technical reader can not only understand but envision.